torch/nn/modules/loss.py

```diff
@@ -290,6 +290,19 @@ class KLDivLoss(_Loss):
     .. _Kullback-Leibler divergence:
         https://en.wikipedia.org/wiki/Kullback-Leibler_divergence
 
+    .. note:: The default averaging means that the loss is actually **not** the
+          KL Divergence because the terms are already probability weighted.
+          A future release of PyTorch may move the default loss closer to the
+          mathematical definition.
+
+          To get the real KL Divergence, use `size_average=False`, and
+          then divide the output by the batch size.
```

Comment:
Can we please get a code example for this? Also it’s not just loss / batch_size, it’s loss.sum() / batch_size

Reply:
`size_average=False` already performs the reduction, so no need to use the `.sum()`

Reply:
So I updated the example to have `batch_size` and actual output.

Reply:
Uh I confused `size_average` with `reduce`! Thanks @fmassa